doc/03_research/03_planning/Implementation.md

```diff
@@ -0,0 +1,135 @@
+
+Lanelet Model Example :
+![Lanelet Model Example](../../00_assets/Lanelets.png)[[Source]](https://github.com/ll7/psaf2/tree/main/Planning/global_planner)
+
+
+Input: 
+* Odometry data (sensoring)
+* GNUU data (sensoring)
+* Map
+* Navigation Waypoints
+
+Output:
+* Route (Sequences of Lanelets and Points) (local path planning, decision making)
+
+---
+
+## **Local Path Planning**
+
+Local Planner updates the current route, if Decision Making detects an obstacle. The planner then has to choose an alternative next Lanelet to avoid the obstacle. Therefore the obstacle position on the lanelet plan will be needed. 
+
+### <u>_Velocity profile_</u>
+
+The Local Path Planer receives the lanelets, points and the path to drive. The local planner creates a velocity profile on the calculated trajectory based on curvature, crossings and traffic lights. This will be calculated directly after the preplanning created a trajectory. The velocity value is published to the acting side, where the current speed and the braking distance is held and calculated. If the velocity is too high or too low, the Acting will realize and sends changes to adapt the speed of the vehicle.
+
+Input:
+
+* Trajectory points (preplanning)
+
+Output:
+
+* Max. Velocity (Acting)
+
+
+### <u>_Update path_</u>
+
+The update path module receives a command from the decision maker, that the current path is not possible to drive without collision. It uses the lanelet modell from the preplanner and chooses the next possible lanelet and accordingly updates the trajectory. It also tells the velocity profile to update because of the new trajectory.
+
+Input:
+
+* lanelet modell (preplanning)
+* update command (decision making)
+* information about blocked lanelets (decision making, perception)
+
+Output:
+* updated trajectory (acting, decision making)
+* update command (velocity profile)
+
+### <u>_Measure distance_</u>
+
+This module measures the distance to obstacles, especially cars, with the Lidar Sensor. The current distance value is published to the acting side for keeping a safe distance (Adaptive Cruise Control).
+
+Input:
+
+* Lidar Sensor data (perception, sensoring)
+
+Output:
+
+* distance value (acting)
+
+---
+## **Decision Making**
+
+Obstacle detection is based on the sensor data from the perception area. If an obstacle is being recognized, the decision making sends a message to the local path planning where the system then chooses another trajectory/lanelet.
+With the Lanelets Model it is possible to give a prediction for other objects and the vehicle itself, by following the lanelet direction of an object.
```

Comment:
Hier vielleicht anmerken, dass das aber nicht im Planning passiert, sondern in der Perception.